graph/pull.go

```diff
@@ -516,15 +524,19 @@ func (s *TagStore) pullV2Tag(r *registry.Session, out io.Writer, endpoint *regis
 
 				if !verifier.Verified() {
 					logrus.Infof("Image verification failed: checksum mismatch for %q", di.digest.String())
+					out.Write(sf.FormatProgress(stringid.TruncateID(img.ID), "Download complete, Checksum failed", nil))
+					di.validChecksum = false
```

Comment:
"verified" represents the image image manifest function coming from "loadManifest".  The verified variable is serving as a global validator for all layers.  If one layer fails during download then it gets flipped to false.  I believe this is the behavior that is expected.  The next best alternative may be to be more verbose and iterate the download layers to determine which has failed if the "verified" boolean is true.